framework/src/modules/interoperability/mainchain/commands/sidechain_registration.ts

```diff
@@ -12,25 +12,281 @@
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_REGISTERED_NAMES,
+		);
+		const nameExists = await nameSubstore.has(Buffer.from(name, 'utf8'));
+
+		if (nameExists) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Name substore must not have an entry for the store key name'),
+			};
+		}
+
+		const networkID = hash(Buffer.concat([genesisBlockID, transaction.senderAddress]));
+
+		// 	networkId has to be unique with respect to the set of already registered sidechain network IDs in the blockchain state.
+		const networkIDSubstore = context.getStore(
+			MODULE_ID_INTEROPERABILITY,
+			STORE_PREFIX_REGISTERED_NETWORK_IDS,
+		);
+		const networkIDExists = await networkIDSubstore.has(networkID);
+
+		if (networkIDExists) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Network ID substore must not have an entry for the store key networkID'),
+			};
+		}
+
+		let totalBftWeight = BigInt(0);
+		for (let i = 0; i < initValidators.length; i += 1) {
+			const currentValidator = initValidators[i];
+
+			// The blsKeys must be lexigraphically ordered and unique within the array.
+			if (
+				initValidators[i + 1] &&
+				currentValidator.blsKey.compare(initValidators[i + 1].blsKey) > -1
+			) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Validators blsKeys must be unique and lexigraphically ordered'),
+				};
+			}
+
+			if (currentValidator.bftWeight <= BigInt(0)) {
+				return {
+					status: VerifyStatus.FAIL,
+					error: new Error('Validator bft weight must be greater than 0'),
+				};
+			}
+
+			totalBftWeight += currentValidator.bftWeight;
+		}
+
+		if (totalBftWeight > MAX_UINT64) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error(`Validator bft weight must not exceed ${MAX_UINT64}`),
+			};
+		}
+
+		// Minimum certificateThreshold value: floor(1/3 * totalWeight) + 1
+		// Note: BigInt truncates to floor
+		if (certificateThreshold < totalBftWeight / BigInt(3) + BigInt(1)) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate threshold below minimum bft weight '),
+			};
+		}
+
+		// Maximum certificateThreshold value: total bft weight
+		if (certificateThreshold > totalBftWeight) {
+			return {
+				status: VerifyStatus.FAIL,
+				error: new Error('Certificate threshold above maximum bft weight'),
+			};
+		}
+
+		return {
+			status: VerifyStatus.OK,
+		};
+	}
+
+	public async execute(context: CommandExecuteContext<SidechainRegistrationParams>): Promise<void> {
+		const {
+			transaction,
+			params: { certificateThreshold, initValidators, genesisBlockID, name },
+			getStore,
+		} = context;
+
+		const networkID = hash(Buffer.concat([genesisBlockID, transaction.senderAddress]));
+
+		// Add an entry in the chain substore
+		const chainSubstore = getStore(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA);
+
+		// Find the latest chainID from db
+		const start = intToBuffer(0, 4, BIG_ENDIAN);
```

Comment:
BIG_ENDIAN is default so you can also remove it

Reply:
start/end maybe more easily do like
```
const start = Buffer.alloc(4, 0);
const end = Buffer.alloc(4, 255);
```

Reply:
it's just optional! other way works =)